counter/src/edit.js

```diff
@@ -0,0 +1,125 @@
+import { __ } from '@wordpress/i18n';
+import { InspectorControls, useBlockProps, RichText } from '@wordpress/block-editor';
+import { useState, useEffect } from '@wordpress/element';
+import { PanelBody, BaseControl, TextControl } from '@wordpress/components';
+
+import './editor.scss';
+
+export default function Edit({ attributes, setAttributes }) {
+	// The block attributes
+	const [pre, setPre] = useState(attributes.pre);
+	const [post, setPost] = useState(attributes.post);
+	const [start, setStart] = useState(attributes.start);
+	const [end, setEnd] = useState(attributes.end);
+	const [duration, setDuration] = useState(attributes.duration);
```

Comment:
Can we just use setAttributes instead of combining useState & setAttribues? I think that this will make the code a bit simpler.

For example:

```
// Get block props
	const blockProps = useBlockProps();
	const { pre, post, start, end, duration } = attributes;
```

and then:

```
	<RichText
	id="counter-pre"
	value={pre}
	onChange={(value) => setAttributes({ pre: value })}
	placeholder={__('eg. Miles traveled ', 'counter')}
	preserveWhiteSpace={true}
	tagName="span"
/>
```

Reply:
This has now been replaced

![image](https://github.com/user-attachments/assets/5e48f2a2-5b2d-400c-9753-5e188c17b055)

![image](https://github.com/user-attachments/assets/9846ce8a-674c-4609-8728-27c0cafa4d2a)